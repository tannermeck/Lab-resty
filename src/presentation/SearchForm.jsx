import React from 'react';

export const SearchForm = ({ search, submit, radioChange, jsonTextField }) => {
    return (
        <>
                <form onSubmit={submit}>
                    <input onChange={search} placeholder="URL"/>
                        <div>
                            <label name="search">GET
                                <input name="search" value="GET" type="radio" onChange={radioChange}/>
                            </label>
                            <label name="search">POST
                                <input name="search" value="POST" type="radio" onChange={radioChange}/>
                            </label>
                            <label name="search">PUT
                                <input name="search" value="PUT" type="radio" onChange={radioChange}/>
                            </label>
                            <label name="search">PATCH
                                <input name="search" value="PATCH" type="radio" onChange={radioChange}/>
                            </label>
                            <label name="search">DELETE
                                <input name="search" value="DELETE" type="radio" onChange={radioChange}/>
                            </label>
                            <button>Go!</button>
                        </div>
                    <textarea placeholder="Raw JSON Body" onChange={jsonTextField}/>
                </form>
        </>
        )
}