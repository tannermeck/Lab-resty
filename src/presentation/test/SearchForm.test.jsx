import React from 'react';
import { render } from '@testing-library/react';
import { SearchForm } from '../SearchForm';

describe('should render a search form', () => {
    it('it should render a component containing a form with an input, 5 radio buttons, and a button', () => {
        const { asFragment } = render(
        <SearchForm search='www.url.com' radioChange='GET'/>);
        expect(asFragment()).toMatchSnapshot()
    })
})