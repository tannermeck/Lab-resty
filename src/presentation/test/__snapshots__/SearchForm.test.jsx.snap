// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a search form it should render a component containing a form with an input, 5 radio buttons, and a button 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="URL"
    />
    <div>
      <label
        for="GET"
      >
        GET
        <input
          id="GET"
          name="search"
          type="radio"
          value="GET"
        />
      </label>
      <label
        for="POST"
      >
        POST
        <input
          id="POST"
          name="search"
          type="radio"
          value="POST"
        />
      </label>
      <label
        for="PUT"
      >
        PUT
        <input
          id="PUT"
          name="search"
          type="radio"
          value="PUT"
        />
      </label>
      <label
        for="PATCH"
      >
        PATCH
        <input
          id="PATCH"
          name="search"
          type="radio"
          value="PATCH"
        />
      </label>
      <label
        for="DELETE"
      >
        DELETE
        <input
          id="DELETE"
          name="search"
          type="radio"
          value="DELETE"
        />
      </label>
      <button>
        Go!
      </button>
    </div>
    <textarea
      placeholder="Raw JSON Body"
    />
  </form>
</DocumentFragment>
`;
